cmake/OpenCVModule.cmake

```diff
@@ -573,6 +575,10 @@ macro(ocv_create_module)
     "${OPENCV_CONFIG_FILE_INCLUDE_DIR}/cvconfig.h" "${OPENCV_CONFIG_FILE_INCLUDE_DIR}/opencv2/opencv_modules.hpp"
     ${${the_module}_pch})
 
+  set(OPENCV_MODULE_${the_module}_INTERFACE_DEFS ${OPENCV_MODULE_${the_module}_INTERFACE_DEFS}
+    CACHE INTERNAL "Public compile definitions of ${full_modname} module")
+  target_compile_definitions("${the_module}" PUBLIC ${OPENCV_MODULE_${the_module}_INTERFACE_DEFS})
```

Comment:
@ilya-lavrenov You may add special macro `ocv_target_compile_definitions` into OpenCVUtils.cmake and use it here (follow [this code](https://github.com/Itseez/opencv/blob/3.1.0/cmake/OpenCVUtils.cmake#L85-L107) but exclude "world" logic, check only CMake version to call `add_definitions` or `target_compile_definitions`)